From 43693ff76a884048b3498f33a755b9fe12ad6335 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Tue, 20 Aug 2024 16:47:00 +0300
Subject: [PATCH] [SW-193226] gaudi3: relocate arcmgmt iccm code to die1 spsram

Since ARC1 MGMT consumed 99% of its own ICCM, to free up some
space, we are relocating certian code and data to DIE1 SPSRAM.

This patch define macro of RELOC_FULL_ADDR and RELOC_SIZE to
pass the preboot_is_valid_address_range() while copy the fit
to its destination address.

Change-Id: Id6568677f47f72860372558a885783b7d905ba59
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index ba2e629..1ee9451 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -53,6 +53,8 @@
 #define ARC1_OFFLOAD_DCCM_FULL_BASE    0x0300007FDB0E0000ull
 #define ARC1_OFFLOAD_DCCM_SIZE         SZ_32K
 
+#define ARC1_RELOC_ICCM_FULL_BASE	0x0300007FDB120000ull
+#define ARC1_RELOC_ICCM_SIZE		SZ_64K
 
 #define SRAM_MEM_REGION_BASE		0x10000000
 #define FW_MEM_REGION_BASE		0x20000000
